charms/istio-pilot/README.md

```diff
@@ -48,9 +57,13 @@ If at any point you use the [istioctl](https://istio.io/latest/docs/reference/co
 
 #### Upgrades across more than one minor version
 
-When the charm detects an upgrade across more than one minor version, it will set an `ErrorStatus` and log the reason.  For example, if you have istio-pilot 1.13.0 deployed and you `juju refresh istio-pilot --channel 1.15/stable`, the upgrade-charm event will fail and the charm will go to `ErrorStatus`.  To resolve this error, refresh the charm again back to a supported version (`juju refresh istio-pilot --channel 1.14/stable`).  The charm should recover and successfully upgrade Istio to 1.14.  You can then do a second refresh to get to Istio 1.15.    
+When the charm detects an upgrade across more than one minor version, it will set an `ErrorStatus` and log the reason.  For example, if you have istio-pilot 1.13.0 deployed and you `juju refresh istio-pilot --channel 1.15/stable`, the upgrade-charm event will fail and the charm will go to `ErrorStatus`.  To resolve this error:
+* refresh the charm again back to a supported version (`juju refresh istio-pilot --channel 1.14/stable`)
+* if this refresh does not clear the error state immediately, use `juju resolve istio-pilot/0 --no-retry` to clear the error on the previous refresh event and move on to the next event.  This should lead to the refresh event for 1.14 firing, but it is a race so there could be other events that fire first
```

Comment:
> but it is a race so there could be other events that fire first

are they ?
upgrade is ensure to be fired the first

Reply:
According to [this discussion](https://chat.charmhub.io/charmhub/pl/wdatgrd557baxdec79wf9yjfpe) it would be a race

Reply:
but why do you need no retry here

if you refreshed to the right charm, then you expect to run upgrade hook. Retry means fire the same event, not the same code

Reply:
the `--no-retry` here is to ignore the *previous* refresh.  So if you do:
* `juju refresh --channel something-broken`  --> Charm goes to error
* `juju refresh --channel something-working` --> Charm still in error from `something-broken`
* `juju resolved` --> resolves the error from `something-broken`, but Juju will try and rerun that same event again and go into error for the same reason
* `juju resolved --no-retry` --> resolves the error from `something-broken` and does not rerun that event, moving onto whatever is next (hopefully the refresh to `something-working`, but it could be some other event)

Reply:
@ca-scribner 
that is my point. Juju will resolve the error by firing the same event. But charm code is changed, so, next time it call the code from another charm

you have two charms that both can handle `upgrade` event